src/graph.h

```diff
@@ -274,13 +280,24 @@ struct DependencyScan {
         dep_loader_(state, deps_log, disk_interface, depfile_parser_options),
         dyndep_loader_(state, disk_interface) {}
 
-  /// Update the |dirty_| state of the given node by inspecting its input edge.
+  /// A wrapper for tests that don't care about validation nodes.
```

Comment:
might be better to adjusts the tests than to create a method just for tests. Not sure though.

Reply:
I think you're right, the existence of this wrapper caused me to miss the DynDeps call to RecomputeDirty, which needs to handle newly discovered validation nodes somehow.  I'll update all the tests to pass a NULL validation_nodes, and make it error if it finds validation nodes in that case.